dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MultipleBackJumpsToForwardJumps.java

```diff
@@ -0,0 +1,78 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.CondJump;
+import com.dat3m.dartagnan.program.event.core.Label;
+
+import static java.util.Comparator.comparingInt;
+
+import java.util.List;
+import java.util.function.ToIntFunction;
+
+/*
+    This pass transforms loops to have a single backjump using forward jumping.
+    Given a loop of the form
+
+    L:
+    ...
+    if X goto L
+    ...
+    if Y goto L
+    More Code
+
+    it transforms it to
+
+    L:
+    ...
+    if X goto Forward_to_L
+    ...
+    if Y goto Forward_to_L
+    goto Continue_after_last
+    Forward_to_L:
+    goto L
+    Continue_after_last
+    More Code
+    ...
+*/
+public class MultipleBackJumpsToForwardJumps implements ProgramProcessor {
```

Comment:
This could just be a `FunctionProcessor` or if you want to keep it a `ProgramProcessor`, you need to manually run an `IdReassignment` pass (you can call it in `run`) to restore correct global ids.

Also, the pass name is oddly specific (and I think not fully correct). Just call it `NormalizeLoops`.